client/post-editor/editor-gutenberg-blocks-warning-dialog/index.jsx

```diff
@@ -0,0 +1,143 @@
+	static propTypes = {
+		translate: PropTypes.func,
+		postContent: PropTypes.string,
+		siteId: PropTypes.number,
+		gutenbergUrl: PropTypes.string,
+		switchToGutenberg: PropTypes.func,
+		openPostRevisionsDialog: PropTypes.func,
+	};
+
+	static defaultProps = {
+		translate: identity,
+		postContent: null,
+		siteId: null,
+		gutenbergUrl: null,
+		switchToGutenberg: noop,
+		openPostRevisionsDialog: noop,
+	};
+
+	state = {
+		isDialogVisible: false,
+		forceClassic: false,
+	};
+
+	shouldComponentUpdate( nextProps, nextState ) {
+		return this.state.isDialogVisible !== nextState.isDialogVisible;
+	}
+
+	static getDerivedStateFromProps( props, state ) {
+		const { postContent } = props;
+		const { forceClassic } = state;
+
+		const hasGutenbergBlocks = content => !! content && content.indexOf( '<!-- wp:' ) !== -1;
+
+		const isDialogVisible = ! forceClassic && hasGutenbergBlocks( postContent );
+
+		return {
+			isDialogVisible,
+		};
+	}
+
+	onUseClassic = () => {
+		this.setState( {
+			forceClassic: true,
+		} );
+	};
+
+	onSwitchToGutenberg = () => {
+		const { switchToGutenberg, siteId, gutenbergUrl } = this.props;
+		switchToGutenberg( siteId, gutenbergUrl );
+	};
+
+	onShowDocumentHistory = e => {
+		e.preventDefault();
+		this.props.openPostRevisionsDialog();
+		this.onUseClassic();
+	};
+
+	render() {
+		const { translate } = this.props;
+		const { isDialogVisible } = this.state;
+		const buttons = [
+			{
+				action: 'gutenberg',
+				label: translate( 'Switch to the new editor' ),
+				onClick: this.onSwitchToGutenberg,
+				isPrimary: true,
+			},
+			{
+				action: 'cancel',
+				label: translate( 'Use the classic editor' ),
+				onClick: this.onUseClassic,
+			},
+		];
+		return (
+			<Dialog isVisible={ isDialogVisible } buttons={ buttons } onClose={ this.onUseClassic }>
+				<h1>{ translate( 'This post uses blocks from the new editor' ) }</h1>
+
+				<p>
+					{ translate(
+						'You can continue to edit this post in the Classic Editor, but you may lose some data and formatting.'
+					) }
+				</p>
+
+				<p>
+					{ translate(
+						'You can also check the {{a}}document history{{/a}} and restore a version of the page from earlier.',
+						{
+							components: {
+								//eslint-disable-next-line jsx-a11y/anchor-is-valid
+								a: <a href="#" onClick={ e => this.onShowDocumentHistory( e ) } />,
+							},
+						}
+					) }
+				</p>
+			</Dialog>
+		);
+	}
+}
+
+export default connect(
+	state => {
+		const postContent = getEditorRawContent( state );
+		const siteId = getSelectedSiteId( state );
+		const postId = getEditorPostId( state );
+		const postType = getEditedPostValue( state, siteId, postId, 'type' );
+		const gutenbergUrl = getGutenbergEditorUrl( state, siteId, postId, postType );
+
+		return {
+			postContent,
+			siteId,
+			gutenbergUrl,
+		};
+	},
+	dispatch => ( {
+		switchToGutenberg: ( siteId, gutenbergUrl ) => {
+			return dispatch( setSelectedEditor( siteId, 'gutenberg', gutenbergUrl ) );
+		},
```

Comment:
the `return` here is unexpected/inconsistent with that from `openPostReviewionsDialog`

is it needed? it doesn't appear like we're using it, maybe go back to parity with the other?

```js
switchToGutenberg: ( siteId, gutenbergUrl ) => dispatch( setSelectedEditor( siteId, 'gutenberg', gutenbergUrl ) ),
```

Reply:
Yeah, the `return` is implicit when using the arrow function in one single line, so both were using the `return`.

I decided to explicitly use it here for avoid a very long line (that is split in 2 by eslint), but I guess it's better to have consistency 😉.

Changed in https://github.com/Automattic/wp-calypso/pull/28406/commits/6ce013ba682c0726b4ca3f23d9ebe5db7c0633c7.